processor/marshalprocessor/processor.go

```diff
@@ -0,0 +1,110 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marshalprocessor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type marshalProcessor struct {
+	logger    *zap.Logger
+	marshalTo string
+}
+
+func newMarshalProcessor(logger *zap.Logger, cfg *Config) *marshalProcessor {
+	return &marshalProcessor{
+		logger:    logger,
+		marshalTo: cfg.MarshalTo,
+	}
+}
+
+// ErrStringBodyNotSupported is returned when a log body is a string
+var ErrStringBodyNotSupported = fmt.Errorf("String body not supported")
+
+func (mp *marshalProcessor) processLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	var errors []error
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceLog := ld.ResourceLogs().At(i)
+		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
+			scopeLog := resourceLog.ScopeLogs().At(j)
+			for k := 0; k < scopeLog.LogRecords().Len(); k++ {
+				logRecord := scopeLog.LogRecords().At(k)
+				logBody := logRecord.Body()
+				// If body is a string, skip that log
+				if logBody.Type() == pcommon.ValueTypeStr {
+					errors = append(errors, ErrStringBodyNotSupported)
+					continue
+				}
```

Comment:
I'm curious on this, why are we only skipping string values? Is it OK for other non-map values to be converted? (e.g. bool, int, float)

Reply:
Also, I'm not sure what happens here if you return an error + logs payload. Are all the logs dropped if you return an error for just one record, or do they get passed on anyways?

Reply:
No, it should only allow a map. 

I am also unsure what happens, I assume we want to log errors for the records that can't be converted but allow the others through.

Reply:
I don't think we should be returning these kinds of errors. We should only return an error if there's something incorrect happening. It looks like when the logsFunc gets called, the logs are dropped if there's an error.
https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/processorhelper/logs.go#L44-L56

Reply:
If anything we can just log what happened

Reply:
Yeah, makes sense